INSTRUCTIONS/LOCAL_DEPLOYMENT/README.md

```diff
@@ -0,0 +1,228 @@
+juju add-model finos-legend
+```
+
+## Deploy the Legend Bundle
+When you deploy an application with Juju, the installation code in the charmed operator will run and set up all the resources and environmental variables needed for the application to run properly. In the case of this tutorial, we are deploying a *bundle*, which describes applications to be deployed and relationships between them.
+
+Deploy the finos-legend-bundle in the finos-legend model using the command line :
+``` bash
+juju deploy finos-legend-bundle --trust --channel=edge
+```
+
+In another terminal window, you can see the applications being deployed and the integration code running
+``` bash
+watch --color juju status --color
+```
+
+You'll notice that the Unit `gitlab-integrator/0` will get to `blocked` status; this is expected, as you'll need to [Setup and Configure GitLab](#Setup-and-Configure-GitLab).
+
+## Setup and Configure GitLab
+To run Legend, you need to either run a GitLab instance somewhere, or use GitLab.com; the type of installation really depends on user's requirements, there is a secion in `DEPLOY_GITLAB.md` that talks about that (TODO).
+
+If this is your first experience with Legend, we suggest you starting with GitLab.com as per the instructions below. If, instead, you're interested to test a Legend deployment with a local GitLab, please follow instructions on `DEPLOY_GITLAB.md`.
+
+If you are using gitlab.com, follow the following three steps.
+
+#### 1. Create a GitLab Application
+
+1. Create an account on gitlab.com
+2. Create an application:
+- Login GitLab.com, click your profile picture on the right upper corner and click "Preferences". 
+- Click "Applications" on the left menu. 
+- Create a new application with the following information:
+  - name
+  - Check the `Confidential` checkbox 
+  - Enter the following Redirect URIs:
+      ```
+      http://legend-studio/studio/log.in/callback
+      http://legend-engine/callback
+      http://legend-sdlc/api/auth/callback
+      http://legend-sdlc/api/pac4j/login/callback
+      ```   
+  - enable the following scopes: 
+    - API
+    - Open ID
+    - Profile
+- Click "Save Application".
+
+On the following page, make a note of the `Application ID` and `Secret`. 
+
+#### 2. Pass the application details to the Legend stack
+
+In your terminal run the following command to pass the `Application ID` and `Secret` to the Legend stack
+``` bash
+juju config gitlab-integrator gitlab-client-id="<Application ID>" gitlab-client-secret="<Secret Id>"
+```
+
+#### 3. Update the GitLab application URIs
+
+The command below will retrieve the GitLab URIs from the GitLab Integrator charm  
+``` bash
+juju show-unit --app gitlab-integrator/0 | python3 -W ignore -c "import json, sys, yaml; doc = yaml.load(sys.stdin); urls = [json.loads(r['application-data']['legend-gitlab-redirect-uris']) for r in doc['gitlab-integrator/0']['relation-info']]; print('\n'.join([url for url_list in urls for url in url_list]))"
+```
```

Comment:
Update to:

```
The command bellow will output the GitLab URIs expected to be used by Legend:

```bash
juju run-action gitlab-integrator/0 get-redirect-uris --wait
``
```